### PR TITLE
Add Java example to sigopt-examples

### PR DESCRIPTION
**Pull request by pfhayes**
@sc932 @alexandraj777

### CODE REVIEW

File: java/src/main/java/com/example/App.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            System.out.println("Computing result for suggestion: " + suggestion.toString());
+            Result result = App.computeResult(suggestion);
+            experiment.report()
+                .addParam("data", new Observation.Builder()
+                    .assignments(suggestion.getAssignments())
+                    .value(result.value)
+                    .valueStddev(result.stddev)
+                    .build())
+                .call();
+            Thread.sleep(1000);
```
Comment (sc932) [inline java/src/main/java/com/example/App.java]: I think this is worth a comment and it probably belongs in the evaluate function "time consuming and expensive to evaluate"

This helps when we're qualifying customers too. If we can get people to think of us for solving "harder" problems only.

File: java/src/main/java/com/example/App.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    }
+}
+
+public class App
+{
+    public static void main(String[] args) throws APIException, InterruptedException
+    {
+        String clientId = null;
+        for (int i = 0; i < args.length; i++) {
+            if (args[i].equals("--client_token") && i < args.length - 1) {
```
Comment (sc932) [inline java/src/main/java/com/example/App.java]: does java not have an argparse equivalent?

Comment (pfhayes) [inline java/src/main/java/com/example/App.java]: There are some third party ones, but for just these few arguments this was easy enough

Comment (sc932) [inline java/src/main/java/com/example/App.java]: sounds good

**Pull request merged**